pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -424,16 +427,13 @@ func buildAutoMtlsSettings(
 		if len(subjectAltNamesToUse) == 0 {
 			subjectAltNamesToUse = serviceAccounts
 		}
-
 		// For backward compatibility, use metadata certs if provided.
-		if proxy.Metadata.TLSClientRootCert != "" {
-			return buildMutualTLS(subjectAltNamesToUse, sniToUse, proxy), userSupplied
+		if hasMetadataCerts(proxy.Metadata) {
+			return buildMutualTLS(tls.SubjectAltNames, tls.Sni, proxy), userSupplied
```

Comment:
why using tls.subjectAltNamesToUse and not subjectAltNamesToUse?

Reply:
Because the file based certs can not use service accounts.

Reply:
Subject Alt names are checked on what is embedded in certs (atleast most cases that I know of including us).  I can add them back if it helps.
But SNI is built for IstioMTLS case which is not relevant for MUTUAL TLS case.

I can

Reply:
Subject Alt names is the identity of the server we are connecting to, I am not sure I understand how its embedded in our own cert?

Reply:
Not in the client cert - it is embedded on server cert - what I meant is it is typically not in terms of service accounts. We have our own way of identifying services (servers). So we do not create service accounts. And I am not sure if using service accounts associated with service is the right thing for MUTUAL TLS when TLS config does not have it. For ISTIO MTLS I understand it is useful.

Reply:
I am a bit confused. SubjectAltNames is configured the validation context, for how the client is verifying the server cert. So putting it in the server cert doesn't seem to help - the client needs to know what is valid?

The MUTUAL part makes sense, can we add it as a comment?